api_server.py

```
@@ -74,38 +74,52 @@ def format_hours(minutes: Optional[int]) -> Optional[str]:
 
-def get_availability(entity_id: int, table: str, now: datetime) -> Dict:
+def get_availability(
+    entity_id: int, table: str, now: datetime, conn: Optional[sqlite3.Connection] = None
+) -> Dict:
     col = "crew_id" if table == "crew_availability" else "appliance_id"
-    with get_db() as conn:
-        curr = conn.execute(
+
+    def _query(db):
+        return db.execute(
             f"SELECT end_time FROM {table} WHERE {col} = ? AND start_time <= ? AND end_time > ? LIMIT 1",
```

Ruff (0.14.14)

[error] Possible SQL injection vector through string-based query construction

(S608)